src/main/java/graphql/execution/ExecutionContext.java

```diff
@@ -6,20 +6,18 @@
 import graphql.language.OperationDefinition;
 import graphql.schema.GraphQLSchema;
 
-import java.util.ArrayList;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 
 public class ExecutionContext {
 
     private GraphQLSchema graphQLSchema;
-    private ExecutionStrategy executionStrategy;
+    private ExecutionStrategy queryStrategy;
+    private ExecutionStrategy mutationStrategy;
     private Map<String, FragmentDefinition> fragmentsByName = new LinkedHashMap<String, FragmentDefinition>();
     private OperationDefinition operationDefinition;
     private Map<String, Object> variables = new LinkedHashMap<String, Object>();
     private Object root;
-    private List<GraphQLError> errors = new ArrayList<GraphQLError>();
+    private List<GraphQLError> errors = Collections.synchronizedList(new ArrayList<GraphQLError>());
```

Comment:
I'm not sure Collections.synchronizedList() is good here anyway. Thanks, I'll check out CopyOnWriteArrayList